source/features/useful-not-found-page.tsx

```diff
@@ -36,54 +38,59 @@ function parseCurrentURL(): string[] {
 	return parts;
 }
 
-// If the resource was deleted, link to the commit history
-async function addCommitHistoryLink(bar: Element): Promise<void> {
-	const parts = parseCurrentURL();
-	parts[2] = 'commits';
-	const url = '/' + parts.join('/');
-	if (await is404(location.origin + url)) {
+async function getLatestChangeToFile(): Promise<Record<string, any> | void> {
+	const url = new GitHubURL(location.href);
+	const {branch, filePath} = url;
+	if (!branch || !filePath) {
 		return;
 	}
 
-	bar.after(
-		<p className="container mt-4 text-center">
-			See also the {getType()}’s <a href={url}>commit history</a>.
-		</p>,
-	);
-}
-
-// If the resource exists in the default branch, link to it
-async function addDefaultBranchLink(bar: Element): Promise<void> {
-	const parts = getCleanPathname().split('/');
-	const branch = parts[3];
-	if (!branch) {
+	const commitsResponseObject = await api.v4(`
+		repository() {
+			ref(qualifiedName: "${branch}") {
+				target {
+					... on Commit {
+						history(first: 1, path: "${filePath}") {
+							nodes {
+								oid
+							}
+						}
+					}
+				}
+			}
+		}
+	`);
+	const commits = commitsResponseObject.repository.ref.target.history.nodes;
+	if (!commits[0]) {
 		return;
 	}
 
-	const defaultBranch = await getDefaultBranch();
-	if (branch === defaultBranch) {
+	// API v4 doesn't support retrieving a list of changed files for a commit:
+	// https://github.community/t/graphql-api-get-list-of-files-related-to-commit/14047/2
+	const commitInfo = await api.v3(`commits/${commits[0].oid as string}`);
+	const fileInfo = commitInfo.files.find((file: AnyObject) => [file.filename, file.previous_filename].includes(filePath));
+	if (!fileInfo) {
 		return;
 	}
 
-	parts[3] = defaultBranch; // Change branch
-	const url = '/' + parts.join('/');
-	if (await is404(location.origin + url)) {
+	return {fileInfo, commitInfo};
```

Comment:
```js
if (fileInfo) return {fileInfo, commitInfo}
```

I'd probably just drop the `.find` as well and just use something like:

```js
for (fileInfo of commitInfo.files) {
	if ([file.filename, file.previous_filename].includes(filePath)) {
		return {fileInfo, commitInfo}
	}
}
```

Reply:
Ok, sure